tardis/rest/app/security.py

```diff
@@ -1,127 +1,119 @@
+    # TODO: Set this too to True. But as soon as possible.
+    # Update: As 'same_site' is strict this is probably enough.
+    authjwt_cookie_csrf_protect: bool = False
 
 
-class UserCredentials(BaseModel):
-    user_name: str
-    hashed_password: str
-    scopes: List[str]
+@AuthJWT.load_config
+def get_config():
+    return Settings()
 
-    class Config:
-        extra = "forbid"
 
+class BaseUser(BaseModel):
+    user_name: str
+    scopes: Optional[List[str]] = None
 
-oauth2_scheme = OAuth2PasswordBearer(
-    tokenUrl="login/access-token",
-    scopes={
-        "resources:get": "Allows to read resource database",
-        "resources:put": "Allows to update resource database.",
-    },
-)
 
+# TODO: Document the scopes manually
+# "resources:get": "Allows to read resource database",
+# "resources:put": "Allows to update resource database.",
 
-def create_access_token(
-    user_name: str,
-    scopes: List[str],
-    expires_delta: Optional[timedelta] = None,
-    secret_key: Optional[str] = None,
-    algorithm: Optional[str] = None,
-) -> str:
-    to_encode = {"sub": user_name, "scopes": scopes}
-    if expires_delta:
-        expire = datetime.utcnow() + expires_delta
-        to_encode.update({"exp": expire})
-    encoded_jwt = jwt.encode(
-        to_encode,
-        secret_key or get_secret_key(),
-        algorithm=algorithm or get_algorithm(),
-    )
 
-    return encoded_jwt
+class LoginUser(BaseUser):
+    password: str
 
 
-def check_authorization(
-    security_scopes: SecurityScopes, token: str = Depends(oauth2_scheme)
-) -> TokenData:
-    if security_scopes.scopes:
-        authenticate_value = f'Bearer scope="{security_scopes.scope_str}"'
-    else:
-        authenticate_value = "Bearer"
+class DatabaseUser(BaseUser):
+    hashed_password: str
 
-    try:
-        payload = jwt.decode(token, get_secret_key(), algorithms=[get_algorithm()])
-        user_name: str = payload.get("sub")
-        token_scopes = payload.get("scopes", [])
-        token_data = TokenData(scopes=token_scopes, user_name=user_name)
-    except (JWTError, ValidationError) as err:
-        raise HTTPException(
-            status_code=status.HTTP_401_UNAUTHORIZED,
-            detail="Could not validate credentials",
-            headers={"WWW-Authenticate": authenticate_value},
-        ) from err
 
-    for scope in security_scopes.scopes:
-        if scope not in token_data.scopes:
+def check_scope_permissions(requested_scopes: List[str], allowed_scopes: List[str]):
+    for scope in requested_scopes:
+        if scope not in allowed_scopes:
             raise HTTPException(
-                status_code=status.HTTP_401_UNAUTHORIZED,
-                detail="Not enough permissions",
-                headers={"WWW-Authenticate": authenticate_value},
+                status_code=status.HTTP_403_FORBIDDEN,
+                detail={
+                    "msg": "Not enough permissions",
+                    "scopesRequired": requested_scopes,
```

Comment:
Should this not be the scope that was requested but not in the `allowed_scopes`?

Reply:
No, it's correct as is. The naming of the parameters is very unfortunate. That's because the function is used in two different contexts. I wanted to prevent code duplication. I'm open for naming suggestions.